utils/src/main/scala/za/co/absa/enceladus/utils/udf/UDFLibrary.scala

```diff
@@ -34,65 +34,74 @@ object UDFLibrary {
 
   private var isRegistered = false
 
-  private def registerUdf()(implicit spark: SparkSession) {
+  private var registry: Set[Int] = Set.empty
+
+  private def shouldRegister(forSpark: SparkSession): Boolean = {
+    var result = false
```

Comment:
Could this variable be removed so we just use the registry variable?

Reply:
I don't know how. The only option would be to place the whole registration into the `synchronized` block. I prefer using a `var` to a long `synchronized` section.